### ISSUE TITLE
sql: privileges can be granted to nonexistent users.

### ISSUE THREAD

**Issue by mberhault** [cockroachdb/cockroach#20226]
eg:
```
root@:26257/system> create database test;
CREATE DATABASE

Time: 11.033915ms

root@:26257/system> grant create on database test to marc;
GRANT

Time: 5.974198ms

root@:26257/system> show users;
+----------+
| username |
+----------+
+----------+
(0 rows)

Time: 1.963447ms

root@:26257/system> drop user marc;
pq: cannot drop users marc: grants still exist on test
```

The `DROP USER` error is due to the check that you cannot remove a user while it still has privileges. Conversely, we should not allow granting privileges to nonexistent users.

**Comment by mberhault** [comment]
I'll be poking at some of this logic for roles-related stuff, so I can take it unless someone else wants it.

**Comment by pananc** [comment]
@mberhault   can you help me review my patch, thanks.

**Comment by mberhault** [comment]
Fixed in #20438.
